### ISSUE TITLE
Add to F-Droid?

### ISSUE THREAD

**Issue by jawz101** [brodeurlv/fastnfitness#59]
Have you considered offering this on F-Droid?

@Izzysoft

**Comment by IzzySoft** [comment]
Similar to your issue on the other project: open an [RFP](https://gitlab.com/fdroid/rfp/issues). Tags are already there (but no APKs, so I cannot take it into my repo straight away), license is fine, verionCode/versionName are missing from gradle (but are available in the Manifest).

**Comment by brodeurlv** [comment]
Hi guys, no problem to put it on F-droid. On the contrary. So should I publish the apk also on github, is that it? Thanks.

**Comment by IzzySoft** [comment]
Thanks, @brodeurlv! F-Droid builds from source (and thus wouldn't use the APK). But until your app is listed there, it might take a little. Providing the APK here (preferably attached to its corresponding tag) would enable me to immediately take it into [my repo](https://apt.izzysoft.de/fdroid/repo) until then (and fetch updates automatically within 24h of the next tag being made with an APK attached). Just let me know when it's there then.

@jawz101 will you file the RFP in the official repo then? All should be ready for that (and more we know as soon as the bot scanned here which won't happen before the RFP is filed :wink:)

**Comment by brodeurlv** [comment]
I'm working on a new release. I will make it clean with the tag and the apk. I will let you know. Maybe this week end, or the next one. 
Thanks

**Comment by IzzySoft** [comment]
Thanks! That looks great! Haven't yet checked whether you use some "blobs" in the source repo (JARs or the likes). If not, I'd say you should go for inclusion with the official repo. I gladly assist! Until then, I've included your app [with my repo](https://apt.izzysoft.de/fdroid/index/apk/com.easyfitness). Enjoy!

**Comment by brodeurlv** [comment]
Thanks izzy.
However I see version 0.16.3 on your site while it should be 0.17.
Did you get this version from the apk?
Thanks

**Comment by IzzySoft** [comment]
Could you please check what the APK says (I can't at the moment as I'm not at my machine)? I don't compile myself, I only pick the APK you attach to the releases. And as there's only one, I hardly could have picked the wrong one :wink:

**Comment by brodeurlv** [comment]
I don't know exactly what went wrong but anyway, I have regenerated the 0.17 apk and the tag. 
It should be good this time, I have unzipped the APK to make sure it is the correct version inside. 
Sorry for that. 
Thanks.

**Comment by brodeurlv** [comment]
@IzzySoft ,
How does it work when I release a new version ? Should I send you a request each time or is it automatic ?
I have just fixed a major issue #60 this morning on recent android version and release a 0.17.1.  
Thanks.

**Comment by IzzySoft** [comment]
It's automatic. Within 24h of your attaching an APK to a new tag it should be grabbed by my updater (and usually is; if you get a report it's not, a ping cannot hurt :wink:). So 0.17.1 should be available via my repo in about 7..8 hours (with the sync around 8pm CET). Thanks!

**Comment by IzzySoft** [comment]
Looks like it worked out, I just checked. Thanks again!

**Comment by brodeurlv** [comment]
I have just created the request for the main repo. 
https://gitlab.com/fdroid/rfp/issues/982
Hope it's ok. 

Thanks.

**Comment by IzzySoft** [comment]
Thank you! Yes, just saw that. Will wait for the bot to check and comment then on its findings (e.g. by posting the metadata for our "integrators" to use).

**Comment by brodeurlv** [comment]
I have removed the  javacsv.jar. It now compiles from source code.
It should fix the issue on FDroid.

**Comment by IzzySoft** [comment]
Thanks! I've asked relan to give it another try.

PS: Should that be a pre-release, or was it accidentally marked such?

**Comment by brodeurlv** [comment]
Thanks IzzySoft. 
I marked it has pre-release has I'm waiting for some beta tester feedback before making it a release and pushing it in production in the google play. 
Does it has an impact on F-Droid ?
Thanks

**Comment by IzzySoft** [comment]
My updater won't pick it as long as it's a pre-release (that is, unless I change that setting for your app). So fine with me, just wanted to make sure. As soon as the "pre-release mark" is gone, my updater will fetch it.

Not sure what impact it would have with the official repo (I've no build experience so I don't know much about that part). But as your app isn't yet listed there, it won't have any impact currently :smile:

**Comment by brodeurlv** [comment]
Ok thanks. I received the feedback I was waiting for. I have changed it to "release" so your repo should be able to take it now. 
Thank you for your help here ! Much appreciated.

**Comment by IzzySoft** [comment]
Thanks! Missed it by ~10 minutes, the update job is just through for today. Will be picked tomorrow then. If you want pre-releases picked as well: no big deal, just need to switch a boolean for that. Tell me, and I do it. Or tell me not to, and I leave it :smile:

**Comment by brodeurlv** [comment]
Hi @IzzySoft,
I was trying to find my app on your repo but I couldn't. This link  https://apt.izzysoft.de/fdroid/index/apk/com.easyfitness, that you provided before in this thread, is heading to a dead end. 
I also could not find it on FDroid.  Can you help me here?
Thank you.

**Comment by IzzySoft** [comment]
Ouch! No idea what happened there – for some reason it got disabled. Hopefully fixed it now, should become visible with the next sync. Will probably trigger one manually to make up for the mess. Link is correct and will work once the sync is complete. Apologies for the confusion!

**Comment by IzzySoft** [comment]
Congrats for reaching the official repo, @brodeurlv! Do you want me to keep the app in mine (faster updates), or shall I rather remove it (to avoid confusion with "signature mismatch")? Usually, at this stage I do the latter. But with the official repo currently being a bit behind (no fresh index for about a week already), I could postpone removal for a little if you wish.

**Comment by brodeurlv** [comment]
Thanks Izzy. Honestly, I'm fine whatever way you prefer. I understand if
you need to clean up your repo.
Anyway, thank you very much for your support during this step to the
official repo. Your availability and kindness has been very appreciated.
Thank you.

**Comment by IzzySoft** [comment]
Glad to! I'll drop it then as soon as the new index (with the 18.1 release) is pushed out. Should the need arise, I can re-establish it anytime, just drop me a note then.
